Investigate TestProxyChecks intermittent failure

```
--- FAIL: TestProxyChecks (22.20s)
	assertions.go:239: 
                          
	Error Trace:	proxy_checks_test.go:99
		
	Error:      	An error is expected but got nil.
		
	Messages:   	{
		
	            	  "timestamp": 1519231750,
		
	            	  "entity": {
		
	            	    "class": "agent",
		
	            	    "deregister": false,
		
	            	    "deregistration": {},
		
	            	    "environment": "default",
		
	            	    "id": "TestProxyChecks",
		
	            	    "keepalive_timeout": 10,
		
	            	    "organization": "default",
		
	            	    "redact": [
		
	            	      "password",
		
	            	      "passwd",
		
	            	      "pass",
		
	            	      "api_key",
		
	            	      "api_token",
		
	            	      "access_key",
		
	            	      "secret_key",
		
	            	      "private_key",
		
	            	      "secret"
		
	            	    ],
		
	            	    "subscriptions": [
		
	            	      "test",
		
	            	      "entity:TestProxyChecks"
		
	            	    ],
		
	            	    "system": {
		
	            	      "hostname": "travis-job-4a935945-eee3-40f3-aeaa-00b876b5fafe",
		
	            	      "os": "linux",
		
	            	      "platform": "ubuntu",
		
	            	      "platform_family": "debian",
		
	            	      "platform_version": "14.04",
		
	            	      "network": {
		
	            	        "interfaces": [
		
	            	          {
		
	            	            "name": "lo",
		
	            	            "addresses": [
		
	            	              "127.0.0.1/8"
		
	            	            ]
		
	            	          },
		
	            	          {
		
	            	            "name": "eth0",
		
	            	            "mac": "42:01:0a:14:01:cc",
		
	            	            "addresses": [
		
	            	              "10.20.1.204/32"
		
	            	            ]
		
	            	          },
		
	            	          {
		
	            	            "name": "docker0",
		
	            	            "mac": "02:42:97:1e:53:63",
		
	            	            "addresses": [
		
	            	              "172.17.0.1/16"
		
	            	            ]
		
	            	          }
		
	            	        ]
		
	            	      },
		
	            	      "arch": "amd64"
		
	            	    },
		
	            	    "user": "agent"
		
	            	  },
		
	            	  "check": {
		
	            	    "check_hooks": null,
		
	            	    "command": "command",
		
	            	    "duration": 0.000868417,
		
	            	    "environment": "default",
		
	            	    "executed": 1519231750,
		
	            	    "handlers": [],
		
	            	    "high_flap_threshold": 0,
		
	            	    "history": null,
		
	            	    "interval": 1,
		
	            	    "low_flap_threshold": 0,
		
	            	    "name": "check_proxy",
		
	            	    "organization": "default",
		
	            	    "proxy_entity_id": "",
		
	            	    "proxy_requests": {
		
	            	      "entity_attributes": [
		
	            	        "entity.Class == \"proxy\""
		
	            	      ]
		
	            	    },
		
	            	    "publish": true,
		
	            	    "round_robin": false,
		
	            	    "runtime_assets": [
		
	            	      "ruby-2-4-2"
		
	            	    ],
		
	            	    "state": "passing",
		
	            	    "stdin": false,
		
	            	    "subdue": null,
		
	            	    "subscriptions": [
		
	            	      "test"
		
	            	    ],
		
	            	    "timeout": 0,
		
	            	    "total_state_change": 0
		
	            	  }
		
	            	}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -39,9 +39,12 @@ subcommand.
 ### Fixed
 - Fixed a bug in time.InWindow that in some cases would cause subdued checks to
 be executed.
-  be executed.
-- Fixed a bug in the HTTP API where resource names could not contain special characters.
-- Resolved a bug in the keepalive monitor timer which was causing it to erroneously expire.
+- Fixed a bug in the HTTP API where resource names could not contain special
+characters.
+- Resolved a bug in the keepalive monitor timer which was causing it to
+erroneously expire.
+- Resolved a bug in how an executor processes checks. If a check contains proxy
+requests, the check should not duplicately execute after the proxy requests.
 
 ## [2.0.0-alpha.17] - 2018-02-13
 ### Added

--- a/backend/schedulerd/executor.go
+++ b/backend/schedulerd/executor.go
@@ -281,10 +281,12 @@ func processCheck(ctx context.Context, executor Executor, check *types.CheckConf
 		if matchedEntities := matchEntities(entities, check.ProxyRequests); len(matchedEntities) != 0 {
 			if err := executor.publishProxyCheckRequests(matchedEntities, check); err != nil {
 				logger.Error(err)
-			} else {
-				logger.Info("no matching entities, check will not be published")
 			}
+		} else {
+			logger.Info("no matching entities, check will not be published")
 		}
+	} else {
+		return executor.execute(check)
 	}
-	return executor.execute(check)
+	return nil
 }

--- a/testing/e2e/proxy_checks_test.go
+++ b/testing/e2e/proxy_checks_test.go
@@ -93,7 +93,7 @@ func TestProxyChecks(t *testing.T) {
 	output, err = sensuctl.run("event", "info", source, checkProxy.Name)
 	assert.NoError(t, err, string(output))
 
-	// Make sure the agent's entity did not produced an event for our proxy check
+	// Make sure the agent's entity did not produce an event for our proxy check
 	// request, because it shouldn't have matched the entity_attributes
 	output, err = sensuctl.run("event", "info", agent.ID, checkProxy.Name)
 	assert.Error(t, err, string(output))

